awd/helpers.py

```diff
@@ -174,29 +232,120 @@ def create_dss_message_body(
             }
         )
 
-    def delete(
-        self, sqs_base_url: str, queue_name: str, receipt_handle: str
-    ) -> EmptyResponseMetadataTypeDef:
-        """Delete message from SQS queue."""
-        logger.debug("Deleting %s from SQS queue: %s", receipt_handle, queue_name)
+    def delete(self, receipt_handle: str) -> EmptyResponseMetadataTypeDef:
+        """Delete message from SQS queue.
+
+        Args:
+            receipt_handle: The receipt handle of the message to be deleted.
+        """
+        logger.debug("Deleting %s from SQS queue: %s", receipt_handle, self.queue_name)
         response = self.client.delete_message(
-            QueueUrl=f"{sqs_base_url}{queue_name}",
+            QueueUrl=f"{self.base_url}{self.queue_name}",
             ReceiptHandle=receipt_handle,
         )
         logger.debug("Message deleted from SQS queue: %s", response)
         return response
 
+    def process_result_message(
+        self,
+        sqs_message: MessageTypeDef,
+        retry_threshold: str,
+    ) -> None:
+        """Validate and then process an SQS result message based on content.
+
+        Args:
+            sqs_message: An SQS result message to be processed.
+            retry_threshold: The number of times to attempt processing an article.
+        """
+        if (
+            self.valid_result_message_attributes(sqs_message=sqs_message)
+            and self.valid_result_message_body(sqs_message=sqs_message)
+            and sqs_message.get("ReceiptHandle")
```

Comment:
Optionally, these could theoretically be grouped in a `validate_sqs_message()` method.  If so, this might allow for dropping back an indentation in this whole method:

```python
if not self.validate_sqs_message():
    raise InvalidSqsMessageError()

# continue with rest of method...
doi = sqs_message["MessageAttributes"]["PackageID"]["StringValue"]
# ...
# ...
```

Reply:
Sidenote: I wrote this comment _before_ seeing that `InvalidSqsMessageError` is already established!  I think that's a good sign, just a change of pattern.  Instead of:

```python
if ...:
    # work...
else:
    raise InvalidSqsMessageError
```

It could be more like:
```python
if not ...:
    raise InvalidSqsMessageError

# normal work...
```

I think ultimately this can make it easier to read and reason about, because once you've mentally navigated past that `if / raise` potential, you can just focus on the work being performed.